src/V3Randomize.cpp

```diff
@@ -119,8 +176,67 @@ class RandomizeMarkVisitor final : public VNVisitorConst {
             m_baseToDerivedMap[basep].insert(nodep);
         }
     }
+    void visit(AstNodeStmt* nodep) override {
+        VL_RESTORER(m_stmtp);
+        m_stmtp = nodep;
+        iterateChildrenConst(nodep);
+    }
     void visit(AstNodeFTaskRef* nodep) override {
         iterateChildrenConst(nodep);
+        if (nodep->name() == "rand_mode") {
+            AstMethodCall* const methodCallp = VN_CAST(nodep, MethodCall);
+            AstNodeExpr* const fromp = methodCallp ? methodCallp->fromp() : nullptr;
+            if (VN_IS(fromp, ArraySel)) {
+                nodep->v3warn(E_UNSUPPORTED,
+                              "Unsupported: 'rand_mode()' on unpacked array element");
+                return;
```

Comment:
Isn't some cleanup needed?  Do you have a test that uses -Wno-UNSUPPORTED and tries this?

Reply:
Good catch

Reply:
Can you disable `UNSUPPORTED`, though? I don't think so, and there's no test like that.